tiledb/sm/storage_manager/consolidator.cc

```diff
@@ -419,29 +407,135 @@ Status Consolidator::consolidate(
     return st;
   }
 
-  std::vector<URI> to_delete;
-  for (const auto& f : to_consolidate)
-    to_delete.emplace_back(f.uri());
-
-  // Delete old fragment metadata. This makes the old fragments invisible
-  st = delete_fragment_metadata(array_uri, to_delete);
+  // Write vacuum file
+  st = write_vacuum_file(*new_fragment_uri, to_consolidate);
   if (!st.ok()) {
-    delete_fragments(to_delete);
     delete query_r;
     delete query_w;
+    bool is_dir = false;
+    auto st2 = storage_manager_->vfs()->is_dir(*new_fragment_uri, &is_dir);
+    (void)st2;  // Perhaps report this once we support an error stack
+    if (is_dir)
+      storage_manager_->vfs()->remove_dir(*new_fragment_uri);
     return st;
   }
 
-  // Delete old fragments. The array does not need to be locked.
-  st = delete_fragments(to_delete);
-
   // Clean up
   delete query_r;
   delete query_w;
 
   return st;
 }
 
+/*
```

Comment:
It seems unusual to have a block comment in the implementation. How about moving this into the header?